File: lint-arwaky/src-python/contract/command_executor_port.py
"""contract — Port for executing external commands."""

from abc import ABC


from ..taxonomy import FilePath, ResponseData, Timeout, PatternList


class ICommandExecutorPort(ABC):
    """Port for executing shell commands and external processes."""

    def execute_command(
        self,
        command: PatternList,
        working_dir: FilePath = FilePath(value="."),
        timeout: Timeout | None = None,
    ) -> ResponseData:
        """Execute a command and return the response."""
        raise NotImplementedError

    def health_check(self) -> ResponseData:
        """Check the health of the execution transport."""
        raise NotImplementedError

File: lint-arwaky/src-rust/contract/command_executor_port.rs
/// contract — Port for executing external commands.
use crate::taxonomy::{FilePath, PatternList, ResponseData};
use std::time::Duration;

#[async_trait::async_trait]
pub trait ICommandExecutorPort: Send + Sync {
    /// Execute a command and return the response.
    async fn execute_command(
        &self,
        command: PatternList,
        working_dir: FilePath,
        timeout: Option<Duration>,
    ) -> anyhow::Result<ResponseData>;

    /// Check the health of the execution transport.
    async fn health_check(&self) -> anyhow::Result<ResponseData>;
}
